draft-bbs-signatures.md

```diff
@@ -534,75 +569,79 @@ Procedure:
 
-13. r3 = r1 ^ -1 mod q
+11. r3 = r1 ^ -1 mod q
 
-14. A' = A * r1
+12. A' = A * r1
 
-15. Abar = A' * (-e) + B * r1
+13. Abar = A' * (-e) + B * r1
 
-16. D = B * r1 + H_s * r2
+14. D = B * r1 + H_s * r2
 
-17. s' = s + r2 * r3
+15. s' = s + r2 * r3
 
-18. C1 = A' * e~ + H_s * r2~
+16. C1 = A' * e~ + H_s * r2~
 
-19. C2 = D * (-r3~) + H_s * s~ + H_j1 * m~_j1 + ... + H_jU * m~_jU
+17. C2 = D * (-r3~) + H_s * s~ + H_h1 * m~_1 + ... + H_hU * m~_U
 
-20. c = hash(PK || Abar || A' || D || C1 || C2 || ph)
+18. c = hash(PK || Abar || A' || D || C1 || C2 || ph)
 
-21. e^ = e~ + c * e
+19. e^ = e~ + c * e
 
-22. r2^ = r2~ + c * r2
+20. r2^ = r2~ + c * r2
 
-23. r3^ = r3~ + c * r3
+21. r3^ = r3~ + c * r3
 
-24. s^ = s~ + c * s'
+22. s^ = s~ + c * s'
 
-25. for j in (j1, j2,..., jU): m^_j = m~_j + c * msg_j
+23. for j in (1, 2,..., U): m^_j = m~_j + c * msg_j
 
-26. proof = (A', Abar, D, c, e^, r2^, r3^, s^, (m^_j1, ..., m^_jU))
+24. proof = (A', Abar, D, c, e^, r2^, r3^, s^, (m^_j1, ..., m^_jU))
 
-27. return proof
+25. return proof
 ```
 
 ### ProofVerify
 
 This operation checks that a proof is valid for a header, vector of revealed messages (along side their index corresponding to their original position when signed) and presentation header against a public key (PK).
 
 ```
-result = ProofVerify(PK, proof, ph, header, (msg_i1,..., msg_iR), RevealedIndexes, (H_1,..., H_L))
+result = ProofVerify(PK, proof, header, ph,          
+                 [(H_i1, msg_i1), ..., (H_iR, msg_iR)], 
+                 [H_h1, ..., H_hU])
 
 Inputs:
 
 - PK (REQUIRED), an octet string of the form outputted by the SkToPk operation.
 - proof (REQUIRED), an octet string of the form outputted by the ProofGen operation.
-- ph (REQUIRED), octet string.
-- header (OPTIONAL), an optional octet string containing context and application specific information. If not supplied, it defaults to an empty string.
-- msg_i1,..., msg_iR (OPTIONAL), octet strings. The revealed messages in input to ProofGen.
-- RevealedIndexes (OPTIONAL), vector of unsigned integers. Indexes of revealed messages.
-- H_1,..., H_L (OPTIONAL), points of G1. The generators in input to Sign.
+- header (OPTIONAL), an optional octet string containing context and application 
+     specific information. If not supplied, it defaults to an empty string.
+- ph (OPTIONAL), octet string.
+- [(H_i1, msg_i1), ..., (H_iR, msg_iR)] (OPTIONAL), a list of generators and message 
+     pairs corresponding to revealed messages. Generators must be G1 points, and the 
+     messages must be non-zero scalars mod q. If not supplied, it will default to the 
+     empty array ("[]"). The list MUST be ordered based on the index of the 
+     generators.
+- [H_h1, ..., H_hU] (OPTIONAL), a list of points in G1. Generators corresponding to 
+     messages to be kept hidden. If not supplied, the list will default to the empty 
+     array ("[]"). The list MUST be ordered based on the index of the generators.
```

Comment:
```suggestion
- [H_h1, ..., H_hU] (OPTIONAL), a list of points in G1. Generators corresponding to 
     un-revealed messages. If not supplied, the list will default to the empty 
     array ("[]"). The list MUST be ordered based on the index of the generators.
```

I think reveal vs un-reveal is better language than adding in another term "hidden"?

Reply:
Thanks! Addressed